### PR TITLE
Rename 'not_differentiable' to 'non_differentiable'.

### PR DESCRIPTION
**Pull request by gchanan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19431 Get rid of unnecessary matches_jit_signature: True specifications.
* #19430 Make one_hot non-differentiable.
* #19429 Remove 'BoolTensor', 'IndexTensor' from frontend specifications.
* #19428 Have _embedding_bag_dense_backward match JIT signature.
* #19427 Have embedding_dense_backward match JIT signature.
* #19426 Hook up non_differentiability in derivatives.yaml when no autograd function is generated.
* #19425 Move non_differentiable_arg_names from autograd functions to differentiability_info.
* #19424 Stop generating autograd functions for derivatives.yaml entries that only specify output differentiability.
* **#19272 Rename 'not_differentiable' to 'non_differentiable'.**

because it flows better.  Also rename 'not_differentiable_args_names' to 'non_differentiable_arg_names' because arguments isn't plural there.

Differential Revision: [D15003381](https://our.internmc.facebook.com/intern/diff/D15003381)

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]
not non-disagreeable

File: tools/autograd/load_derivatives.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ def load_derivatives(path, declarations):
 
     return autograd_functions
 
+
```
Comment (cpuhrsch) [inline tools/autograd/load_derivatives.py]: Are you sure?

Comment (gchanan) [reply tools/autograd/load_derivatives.py]: let's see what flake8 says.

Comment (cpuhrsch) [reply tools/autograd/load_derivatives.py]: I have strong opinions on this that I'd like to talk to you in person about.

**Pull request closed without merging**